src/tome.pm

```diff
@@ -144,18 +146,19 @@ saved in the array previously and uses it as a return value.
 
 sub message($$) {
    my $self = shift;
-   my $txt = shift // die 'Insufficient arguments';
-
-   my $rndkey = (keys %tome)[rand keys %tome];
+   my $txt  = shift // die 'Insufficient arguments';
    
-   if ($txt =~ m{[^\s\n+\d-]}) {
+   my $rnd_idx = int rand scalar @tome; 
+   my $rnd_msg = $tome[$rnd_idx];
+   
+   if ($txt =~ m{[^\s+\d-]}) {
       my $txt = (split '\n', $txt)[0];
-
-      delete $tome{ $rndkey } if (keys %tome >= $tome_max);
-      $tome{$txt} = substr $txt, 0, $tome_msg_max;
+      $txt = substr ($txt, 0, $tome_msg_max) if length $txt >= $tome_msg_max;
+      delete $tome[$rnd_idx] if scalar @tome >= $tome_max;
```

Comment:
A deletion from an array is a very expensive operation. 
As I mentioned sometime ago the performant way is to write it like this:
```perl
if (@tome >= $tome_max) {
   $tome[$rnd_idx] = $txt;
} else {
   push @tome, $txt;
}
```